docusaurus/docs/guides/form-creation-overrides.md

```diff
@@ -0,0 +1,103 @@
+---
+id: form-creation-overrides
+title: Form creation overrides
+sidebar_label: Form creation overrides
+---
+
+## Override creating relationship form objects
+
+Use `create{propertyName}FormObject` for creating nested forms. For example, `User` model has `address` `BelongsTo` relationship:
+
+```ts title="user.model.ts"
+import { BelongsTo } from 'ngx-form-object';
+
+class User {
+  @BelongsTo()
+  address: Address;
+}
+```
+
+```ts title="address.model.ts"
+import { Attribute } from 'ngx-form-object';
+
+class Address {
+  @Attribute()
+  street: string;
+}
+```
+
+By default, `userForm.controls.address` will be created as `ExtendedFormControl` with `Address` model set as value.
+
+To have a nested form and be able to edit `street` property `userForm.controls.address` should be a `FormGroup` (or `FormStore`) containing `ExtendedFromControl` for `street` form field.
+
+To achive that override create method in `UserFormObject`. This method must have a name formatted like `create{propertyName}FormObject` and return a `FormObject` instance. It receives model and form object options as its arguments.
+
+```ts title="user.form-object.ts"
+public createAddressFormObject(model: Address, options: FormObjectOptions): AddressFormObject {
+  return new AddressFormObject(model, options);
+}
+```
+This results in `userForm.controls.address` being a `FormStore` created out of the `AddressFormObject` which contains `street` as `ExtendedFormControl`.
+
+A similar method can be defined for `HasMany` relationships. You can define the following method for a `User` model that has a `Car` `HasMany` relationship:
+
+```ts title="user.model.ts"
+import { HasMany } from 'ngx-form-object';
+
+class User {
+  @HasMany()
+  cars: Array<Car>;
+}
+```
+
+```ts title="car.model.ts"
+import { Attribute } from 'ngx-form-object';
+
+class Car {
+  @Attribute()
+  color: string;
+}
+```
+
+Then, in `UserFormObject` define create method override:
+
+```ts title="user.form-object.ts"
+public createCarsFormObject(model: Car, options: FormObjectOptions): CarFormObject {
+  return new CarFormObject(model, options);
+}
+```
+For each `Car` model `FormStore` will be created with `CarFormObjects` returned by this method. This will result in `userForm.controls.cars` being an `ExtendedFormArray` populated with this `FormStores`.
```

Comment:
```suggestion
For each `Car` model, `FormStore` will be created with `CarFormObjects` returned by this method. This will result in `userForm.get('cars')` being an `ExtendedFormArray` populated with this `FormStores`.
```